src/gfx_layout.cpp

```diff
@@ -280,17 +282,18 @@ const char *Layouter::GetCharAtPosition(int x) const
 				size_t index = run.GetGlyphToCharMap()[i];
 
 				size_t cur_idx = 0;
-				for (const char *str = this->string; *str != '\0'; ) {
-					if (cur_idx == index) return str;
+				int char_index = 0;
+				for (auto str = this->string.begin(); str < this->string.end(); char_index++) {
```

Comment:
I'm personally not really fond of comparing iterators with `<`. Primarily because the specification does not mention `<` or `>`, only `==` and `!=`. Conceptually, with reverse iterators, linked lists, deques and anything where data might not be retrieved in ascending order, you have to hope it's implemented in a manner that compares the index and not the memory location.

Reply:
Where my scare is, that `Utf8Consume` might consume more bytes that `string` is large, and as such, it never actually hits `end()`.

On Linux, a `std::string_view::iterator` is just a plain `const char *`, and no bound-checks are done when manipulating it.

So I fully agree with you on any other type, that you should never do `< end()`. But here .. I am a bit scared.
But I guess it is unfounded scareness, as we already know the strings are valid Utf8, so it should never happen that it misses `end()`.

So okay, let's be proper, and use `!=`. But I am still a tiny bit scared :D